pip_audit/_dependency_source/pyproject.py

```diff
@@ -33,14 +28,20 @@ class PyProjectSource(DependencySource):
     """
 
     def __init__(
-        self, filename: Path, index_urls: list[str] = [PYPI_URL], state: AuditState = AuditState()
+        self,
+        filename: Path,
+        index_url: str = "",
```

Comment:
Similar typestate nit here: maybe drop the default `""` here (or make it `PYPI_URL`), since we always expect a valid URL?

Reply:
Yep, I'd prefer this too.